llm-complete-guide/pipelines/finetune_embeddings_legacy.py

```diff
@@ -0,0 +1,101 @@
+# Apache Software License 2.0
+#
+# Copyright (c) ZenML GmbH 2024. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from constants import (
+    CHUNKING_METHOD,
+    DATASET_NAME,
+    MODEL_PATH,
+    NUM_EPOCHS,
+    NUM_GENERATIONS,
+    WARMUP_STEPS,
+)
+from steps.chunk_documents import (
+    chunk_documents,
+)
+from steps.finetune_embeddings_legacy import (
+    # dummy_evaluate_model,
+    # dummy_load_datasets,
+    # dummy_train_model,
+    evaluate_model,
+    load_datasets,
+    train_model,
+)
+from steps.generate_questions import generate_questions
+from steps.huggingface_dataset_upload import (
+    upload_chunks_dataset_to_huggingface,
+)
+from steps.markdown_loader import load_markdown_files
+from steps.preprocess_markdown import preprocess_markdown_texts
+from zenml import pipeline
+
+
+@pipeline
+def finetune_embeddings() -> float:
+    """Fine-tunes embeddings and evaluates the model."""
+    train_dataset, test_dataset = load_datasets(DATASET_NAME)
+
+    model = train_model(
+        train_dataset,
+        model_path=MODEL_PATH,
+        num_epochs=NUM_EPOCHS,
+        warmup_steps=WARMUP_STEPS,
+    )
+
+    evaluate_model(model, MODEL_PATH, test_dataset)
+
+
+# # TODO: FOR TESTING ONLY (remove after)
```

Comment:
still wanna keep this one in?

Reply:
yeah it's part of a bigger experiment. It's in the legacy pipeline. I'd leave it for now.